### PR TITLE
Milestone 4: Filter

### PR DESCRIPTION
**Pull request by rukundoeric**
## What does this do? 
*  Filter
## Description of Task to be completed? 
*  In this milestone, I added functionality to filter by selected category.
## How should this be manually tested? 
NOTE: Make sure you have Node.js and npm or yarn installed in your PC.
- Clone this repo: `git clone https://github.com/rukundoeric/bookstore.git`
- Navigate into project directory: `cd bookstore`
- Install all packages: `yarn install or npm install`
- Start the app: `yarn start or npm start`

### CODE REVIEW

File: src/components/containers/BooksForm.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/forbid-prop-types */
 import React, { Component } from 'react';
```
Comment (uche-inyama) [inline src/components/containers/BooksForm.js]: Do not disable the eslint rule. Kindly remove the disable comment on line 1 and fix the issue

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required.

### HI @rukundoeric :wave: 

I can see you disabled an eslint rule, in some of the files in your project. I have left a comment below, which applies to all the files you have a disabled eslint rule.


#### Fix this, then submit another form for a code review.


### Have a great day :slightly_smiling_face: 


#### ``` Uchechukwu Inyama Henry ```

File: src/components/containers/BooksForm.js (hunk 1 above)
Comment (rukundoeric) [reply src/components/containers/BooksForm.js]: But that one can't be fixed, since it is  mandatory in some cases to pass object or array in props

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @rukundoeric :wave: 

Kindly go through this [material](https://reactjs.org/docs/typechecking-with-proptypes.html)

For example if want to validate an objects propType, you can do it like so:

``` 
      // An object taking on a particular shape
      optionalObjectWithShape: PropTypes.shape({
        color: PropTypes.string,
        fontSize: PropTypes.number
  }),
```

Also, for an array, we have: 
```   
// An array of a certain type
  optionalArrayOf: PropTypes.arrayOf(PropTypes.number),
```

These information above, I got it from this [link](https://reactjs.org/docs/typechecking-with-proptypes.html)

#### Make these changes, the submit another form for a code review.


### Have a great day :slightly_smiling_face: 


 #### ``` Inyama Uchechukwu Henry ```

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @rukundoeric :wave: 

Kindly go through this [material](https://reactjs.org/docs/typechecking-with-proptypes.html)

For example if you want to validate an objects propType, you can do it like so:

``` 
      // An object taking on a particular shape
      optionalObjectWithShape: PropTypes.shape({
        color: PropTypes.string,
        fontSize: PropTypes.number
  }),
```

Also, for an array, we have: 
```   
// An array of a certain type
  optionalArrayOf: PropTypes.arrayOf(PropTypes.number),
```

These information above, I got it from this [link](https://reactjs.org/docs/typechecking-with-proptypes.html)

#### Make these changes, the submit another form for a code review.


### Have a great day :slightly_smiling_face: 


 #### ``` Inyama Uchechukwu Henry ```

**Review by Torres-ssf — APPROVED** [review]

File: src/components/containers/BooksList.js [hunk 2]
```diff
@@ -1,47 +1,85 @@
-/* eslint-disable react/forbid-prop-types */
```
Comment (Torres-ssf) [inline src/components/containers/BooksList.js:1]: - [ ] We can safely remove this eslint disable line. Let's remove it before merging the PR

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :clap: :tada: 
Hello @rukundoeric :hand: 
Awesome job implementing the requested changes.

- [ ] Check my comment below.

Good luck with the next milestone.
Happy coding :computer: 
**Sérgio Torres**

**Pull request merged**